qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -398,21 +424,43 @@ def _wait_for_job(self, timeout=60, wait=5):
 
         job_data = self._api.get_job(self._id)
         job_result_list = []
-        for circuit_result in job_data['qasms']:
-            this_result = {'data': circuit_result['data'],
-                           'name': circuit_result.get('name'),
-                           'compiled_circuit_qasm': circuit_result.get('qasm'),
-                           'status': circuit_result['status']}
-            if 'metadata' in circuit_result:
-                this_result['metadata'] = circuit_result['metadata']
-
-            job_result_list.append(this_result)
-
-        return Result({'id': self._id,
-                       'status': job_data['status'],
-                       'used_credits': job_data.get('usedCredits'),
-                       'result': job_result_list,
-                       'backend_name': self.backend_name})
+        if 'result' in job_data:
+            job_result = job_data['result']
+            # TODO: temporary conversion to pre-qobj 1.0 result to avoid
+            # conflict with PR#773
+            for iresult, resultobj in enumerate(job_result['results']):
+                circuit_name = 'circuit{}'.format(iresult)
+                if 'header' in resultobj:
+                    if 'name' in resultobj['header']:
+                        circuit_name = resultobj['header']['name']
+                    compiled_circuit_qasm = resultobj['header'].get('compiled_circuit_qasm')
+                this_result = {'data': {'counts': resultobj.get('data')},
+                               'name': circuit_name,
+                               'compiled_circuit_qasm': compiled_circuit_qasm,
+                               'status': resultobj.get('status')}
+                job_result_list.append(this_result)
+            job_result = {'id': self._id,
+                          'status': job_result.get('status'),
+                          'result': job_result_list}
+            return Result(job_result)
+        elif 'qasms' in job_data:
+            for circuit_result in job_data['qasms']:
+                this_result = {'data': circuit_result['data'],
+                               'name': circuit_result.get('name'),
+                               'compiled_circuit_qasm': circuit_result.get('qasm'),
+                               'status': circuit_result['status']}
+                if 'metadata' in circuit_result:
```

Comment:
Now that `metadata` is only returned for the "old-format", can we have the new format return the `header` field of the result? According to the schema, the result header is the same as the qobj header, unchanged. In the header we can include qobj itself, so that just by looking at the result, one can tell the experiment that was run.

Reply:
It should be doing this already.